lib/ipcc.c

```diff
@@ -366,6 +366,18 @@ qb_ipcc_fd_get(struct qb_ipcc_connection * c, int32_t * fd)
 	return 0;
 }
 
+int32_t
+qb_ipcc_auth_get(struct qb_ipcc_connection * c, pid_t *pid, uid_t *uid, gid_t *gid)
+{
+	if (c == NULL || pid == NULL || uid == NULL || gid == NULL) {
+		return -EINVAL;
+	}
+	*pid = c->server_pid;
+	*uid = c->euid;
+	*gid = c->egid;
```

Comment:
For flexibility, I wouldn't require these to be non-NULL but wrap the assignments in if != NULL. That way callers can get just one or two values if that's all they need.